src/mame/apple/apple2e.cpp

```diff
@@ -3936,36 +4109,36 @@ static INPUT_PORTS_START( apple2e_common )
 
 	PORT_START("X1")
 	PORT_BIT(0x001, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_NAME("Tab")      PORT_CODE(KEYCODE_TAB)      PORT_CHAR(9)
-	PORT_BIT(0x002, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_Q)  PORT_CHAR('Q') PORT_CHAR('q')
-	PORT_BIT(0x004, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_W)  PORT_CHAR('W') PORT_CHAR('w')
-	PORT_BIT(0x008, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_E)  PORT_CHAR('E') PORT_CHAR('e')
-	PORT_BIT(0x010, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_R)  PORT_CHAR('R') PORT_CHAR('r')
-	PORT_BIT(0x020, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_Y)  PORT_CHAR('Y') PORT_CHAR('y')
-	PORT_BIT(0x040, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_T)  PORT_CHAR('T') PORT_CHAR('t')
-	PORT_BIT(0x080, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_U)  PORT_CHAR('U') PORT_CHAR('u')
-	PORT_BIT(0x100, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_I)  PORT_CHAR('I') PORT_CHAR('i')
-	PORT_BIT(0x200, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_O)  PORT_CHAR('O') PORT_CHAR('o')
+	PORT_BIT(0x002, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_Q)  PORT_CHAR('q') PORT_CHAR('Q')
+	PORT_BIT(0x004, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_W)  PORT_CHAR('w') PORT_CHAR('W')
+	PORT_BIT(0x008, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_E)  PORT_CHAR('e') PORT_CHAR('E')
+	PORT_BIT(0x010, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_R)  PORT_CHAR('r') PORT_CHAR('R')
+	PORT_BIT(0x020, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_Y)  PORT_CHAR('y') PORT_CHAR('Y')
+	PORT_BIT(0x040, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_T)  PORT_CHAR('t') PORT_CHAR('T')
+	PORT_BIT(0x080, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_U)  PORT_CHAR('u') PORT_CHAR('U')
+	PORT_BIT(0x100, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_I)  PORT_CHAR('i') PORT_CHAR('I')
+	PORT_BIT(0x200, IP_ACTIVE_HIGH, IPT_KEYBOARD) PORT_CODE(KEYCODE_O)  PORT_CHAR('o') PORT_CHAR('O')
```

Comment:
Why are the `PORT_CHAR` swapped here?  Don’t unshifted keys produce uppercase by default?  Or am I remembering wrong?

Reply:
Hi! Just tested the PR and it works fine. However this is indeed weird to have to toggle Caps Lock to type lowercase. It's inverted with the host computer. The non-french version (apple2c0 for example) does not require me to caps lock to type lowercase.
So I'm wondering if that should be de-inverted?

Reply:
@colinleroy, with both MAME .271 (and earlier as far as I can remember) and the current PR, I see something similar with both apple2c0 and apple2eefr.

The initial default when starting an Apple IIe/IIc machine for the first time has Caps Lock enabled within the emulated machine.  I am not sure if the default being Caps Lock enabled is due to these lines making Caps Lock active-low:

PORT_BIT( 0x01, IP_ACTIVE_LOW,  IPT_KEYBOARD) PORT_NAME("Caps Lock")    PORT_CODE(KEYCODE_CAPSLOCK) PORT_TOGGLE
PORT_BIT( 0x01, IP_ACTIVE_LOW,  IPT_KEYBOARD) PORT_NAME("Shift Lock")    PORT_CODE(KEYCODE_CAPSLOCK) PORT_TOGGLE

With Caps Lock enabled within the emulated machine, pressing Caps Lock on the host machine disables Caps Lock within the running emulator, but enables Caps Lock on the host.  Exiting the emulator seems to track the emulator caps lock state within the emulator's .cfg file.  This line gets added if Caps Lock disabled, otherwise, this line does not exist:
            <port tag=":keyb_special" type="KEYBOARD" mask="1" defvalue="1" value="0" />
If exiting MAME with the Apple IIe/c's Caps Lock disabled, but the host's enabled, Caps lock can then be toggled off on the host with the emulator not running, the emulator can then be restarted, and then Caps Lock will match between the emulator and the host.  Note that there is one .cfg file per machine, so making one Apple IIe/c variant consistent with the host will not automatically make all of them consistent.

Reply:
Thanks.  I was wondering why the XML line I had pasted disappeared.